OrangeBricks.Web/Controllers/Property/Builders/MyViewingsViewModelBuilder.cs

```diff
@@ -0,0 +1,48 @@
+﻿using OrangeBricks.Web.Controllers.Property.ViewModels;
+using OrangeBricks.Web.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OrangeBricks.Web.Controllers.Property.Builders
+{
+    public class MyViewingsViewModelBuilder
+    {
+        private readonly IOrangeBricksContext _context;
+
+        public MyViewingsViewModelBuilder(IOrangeBricksContext context)
+        {
+            _context = context;
+        }
+
+        public MyViewingsViewModel Build(string buyerUserId)
+        {
+            var viewings = _context.Viewings.Where(v => v.BuyerUserId == buyerUserId).ToList();
+
+            return new MyViewingsViewModel
+            {
+                HasViewings = viewings.Any(),
+                Viewings = CreateViewingViewModel(viewings)
+            };
+        }
+
+        private IEnumerable<BookViewingViewModel> CreateViewingViewModel(List<Viewing> viewings)

```

Comment:
Should have been refactored out into separate class, to be used by the BookedViewingsViewModelBuilder and this MyViewingsViewModelBuilder.